Proposal to remove Sprout fields from the V5 transaction format. (#453)
zcash/zips: zip-0225.rst

   T.4b.ii : cmx                  (field encoding bytes)
   T.4b.iii: ephemeralKey         (field encoding bytes)
   T.4b.iv : encCiphertext[..52]  (First 52 bytes of field encoding)

The personalization field of this hash is set to::

  "ZTxIdOrcOutCHash"

T.4c: orchard_actions_memos_digest
""""""""""""""""""""""""""""""""""

A BLAKE2b-256 hash of the subset of Orchard shielded memo field data for all Orchard
Actions belonging to the transaction. For each Action, the following elements are included
in the hash::

   T.4c.i: encCiphertext[52..564] (contents of the encrypted memo field)

The personalization field of this hash is set to::

  "ZTxIdOrcOutMHash"

T.4d: orchard_actions_noncompact_digest
"""""""""""""""""""""""""""""""""""""""

A BLAKE2b-256 hash of the remaining subset of Orchard Action information **not** intended
for inclusion in an updated version of the the ZIP 307 [#zip-0307]_ ``CompactBlock``
format, for all Orchard Actions belonging to the transaction. For each Action,
the following elements are included in the hash::

   T.4d.i  : cv                    (field encoding bytes)
   T.4d.ii : rk                    (field encoding bytes)

>    T.4d.iii: encCiphertext[464..]  (post-memo suffix of field encoding)

Reviewer: ```suggestion
   T.4d.iii: encCiphertext[564..]  (post-memo suffix of field encoding)
```

I think the offset may have been changed unintentionally.

Author: Good catch!

  "ZTxAuthOrchaHash"

Alternatives
============

The original version of ZIP-225 included Sprout-related fields ``nJoinSplit``, 
``vJoinSplit``, ``joinSplitPubKey``, and ``joinSplitSig`` in the V5
transaction format. The Electric Coin Company and Zcash Foundation teams have
elected to remove these fields from the V5 transaction format as part of the 
continuing process of deprecation of the Sprout shielded pool. As a consequence
of these fields being removed::

* This effectively prohibits migration transactions that would directly move funds from
  the Sprout pool to the Orchard pool. Sprout -> Transparent and Sprout -> Sapling
  migration transactions will still be supported when using the V4 transaction format. 

Removing these fields reduces the complexity of the NU5 upgrade in the following ways::

* V5 parsing and serialization code does not need to take these fields into account.
* ZIP 244 [#zip_0244]_ transaction identifier, signature hash, and authorizing
  data commitment computations are simplified by excluding consideration of 
  these fields.

> Having removed these fields means that that in the future, removing the support for the V4
> transaction format will also effectively end support for Sprout transactions on the zcash

Reviewer: ```suggestion
Removal of these fields means that that in the future, removing the support for the V4
transaction format will also effectively end support for Sprout transactions on the Zcash
```

network, though it might be possible to restore limited support for migration via a future

> ZIP 222 [#zip_0222] extension or by other means not yet determined.

Reviewer: ```suggestion
ZIP 222 [#zip-0222]_ extension or by other means not yet determined.
```
Also this reference needs to be added.

  "ZTxAuthOrchaHash"

Alternatives
============

A possible modification to this proposal is to entirely remove the Sprout-related fields
``nJoinSplit``, ``vJoinSplit``, ``joinSplitPubKey``, and ``joinSplitSig`` from the V5
transaction format. While this would effectively prohibit migration transactions that
directly move funds from the Sprout pool to the Orchard pool, Sprout -> Transparent and

> Sprout -> Sapling transactions would still be supported when using the V4 transaction

Reviewer: ```suggestion
Sprout -> Sapling migration transactions would still be supported when using the V4 transaction
```
